src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1039,50 +1039,74 @@ private void RemoveWorseMembers<TMember>(ArrayBuilder<MemberResolutionResult<TMe
 
-            for (int c1Idx = 0; c1Idx < results.Count; c1Idx++)
+            int currentBestIndex = -1;
+            for (int index = 0; index < results.Count; index++)
             {
-                var c1result = results[c1Idx];
-
-                // If we already know this is worse than something else, no need to check again.
-                if (!c1result.Result.IsValid || worse[c1Idx] == worseThanSomething)
+                // find the first applicable candidate
+                if (!results[index].Result.IsValid)
                 {
                     continue;
                 }
 
-                bool c1IsBetterThanEverything = true;
-                for (int c2Idx = 0; c2Idx < results.Count; c2Idx++)
+                // assume that the current applicable candidate is the best if we don't have any
+                if (currentBestIndex == -1)
                 {
-                    var c2result = results[c2Idx];
-
-                    if (!c2result.Result.IsValid)
-                    {
-                        continue;
-                    }
-
-                    if (c1result.Member == c2result.Member)
+                    worse[currentBestIndex = index] = betterThanEverything;
+                }
+                else
+                {
+                    var better = BetterFunctionMember(results[currentBestIndex], results[index], arguments.Arguments, ref useSiteDiagnostics);
+                    if (better == BetterResult.Left)
                     {
-                        continue;
+                        // the current best is better 
+                        worse[index] = worseThanSomething;
                     }
-
-                    var better = BetterFunctionMember(c1result, c2result, arguments.Arguments, ref useSiteDiagnostics);
-                    if (better == BetterResult.Left)
+                    else if (better == BetterResult.Right)
                     {
-                        worse[c2Idx] = worseThanSomething;
+                        // the current best is worse
+                        worse[currentBestIndex] = worseThanSomething;
+                        worse[currentBestIndex = index] = betterThanEverything;
                     }
                     else
                     {
-                        c1IsBetterThanEverything = false;
-                        if (better == BetterResult.Right)
-                        {
-                            worse[c1Idx] = worseThanSomething;
-                            break;
-                        }
+                        // the current best is not better
+                        worse[currentBestIndex] = worse[index] = notBetterThanEverything;
+                        currentBestIndex = -1;
                     }
                 }
-                if (worse[c1Idx] == unknown)
+            }
+
+            // make sure that every candidate up to the current best is worse
+            for (int index = 0; index < currentBestIndex; index++)
```

Comment:
>for (int index = 0; index < currentBestIndex; index++) [](start = 12, length = 54)

Given that we can reach this loop with ```currentBestIndex == -1``` and that we are still breaking out of the loop early, I believe we still have the same issue of executing the rest of the code in this function based on an incomplete analysis. Note, the problem in here is no different from the one we had for binary operators. I am recommending using the same approach - use fast path if we are able to locate single best candidate, otherwise fall back to the full analysis.